### PR TITLE
Moving hipBLAS calls in tests

### PR DESCRIPTION
**Pull request by daineAMD**
Moving all the hipBLAS calls into the `if(unit_check || norm_check)` section, will save some time for hipblas-bench calls when the unit_check/norm_check isn't necessary. See https://github.com/ROCmSoftwarePlatform/hipBLAS/pull/373

### CODE REVIEW

File: clients/include/testing_getrf_npvt.hpp [hunk 1]
```diff
@@ -70,28 +70,26 @@ hipblasStatus_t testing_getrf_npvt(const Arguments& argus)
...
+        status = hipblasGetrfFn(handle, N, dA, lda, nullptr, dInfo);
 
-    // Copy output from device to CPU
-    CHECK_HIP_ERROR(hipMemcpy(hA1.data(), dA, A_size * sizeof(T), hipMemcpyDeviceToHost));
-    CHECK_HIP_ERROR(hipMemcpy(hInfo1.data(), dInfo, sizeof(int), hipMemcpyDeviceToHost));
+        if(status != HIPBLAS_STATUS_SUCCESS)
+        {
+            hipblasDestroy(handle);
+            return status;
+        }
```
Comment (TorreZuk) [inline clients/include/testing_getrf_npvt.hpp:84]: Can the localHandle pattern not be used?

File: clients/include/testing_getrf_npvt_strided_batched.hpp [hunk 2]
```diff
@@ -84,30 +84,28 @@ hipblasStatus_t testing_getrf_npvt_strided_batched(const Arguments& argus)
...
 
-    // Copy output from device to CPU
-    CHECK_HIP_ERROR(hipMemcpy(hA1.data(), dA, A_size * sizeof(T), hipMemcpyDeviceToHost));
-    CHECK_HIP_ERROR(
-        hipMemcpy(hInfo1.data(), dInfo, batch_count * sizeof(int), hipMemcpyDeviceToHost));
+        if(status != HIPBLAS_STATUS_SUCCESS)
+        {
+            hipblasDestroy(handle);
+            return status;
+        }
```
Comment (TorreZuk) [inline clients/include/testing_getrf_npvt_strided_batched.hpp:99]: same here...

File: clients/include/testing_getri_npvt_batched.hpp [hunk 3]
```diff
@@ -96,30 +96,29 @@ hipblasStatus_t testing_getri_npvt_batched(const Arguments& argus)
...
+        if(status != HIPBLAS_STATUS_SUCCESS)
+        {
+            hipblasDestroy(handle);
+            return status;
+        }
+
+        // Copy output from device to CPU
+        for(int b = 0; b < batch_count; b++)
+            CHECK_HIP_ERROR(
+                hipMemcpy(hA1[b].data(), bC[b], A_size * sizeof(T), hipMemcpyDeviceToHost));
```
Comment (TorreZuk) [inline clients/include/testing_getri_npvt_batched.hpp:115]: { } may make it more readable here

**Review by TorreZuk — APPROVED** [review]
LGTM if all tests pass.  Comments can be for later PR.

File: clients/include/testing_getrf_npvt.hpp (hunk 1 above)
Comment (daineAMD) [reply clients/include/testing_getrf_npvt.hpp:84]: Yeah, the solver _npvt tests are new so I missed them when doing the other solver tests. I'll add them to hipblas-bench, update the localHandle, etc. in another PR.

**Pull request merged**